test/helpers/files.spec.ts

```diff
@@ -237,4 +237,13 @@ describe('getSourceFileForPage', () => {
 
     expect(filePath.replace(TEST_DIR, '')).toBe('/fixtures/page-extensions/custom/pages/api/custom.api.js')
   })
+
+  it('handles getting file when index of folder', () => {
+    const pagesDir = resolve(__dirname, '../fixtures/page-extensions/index/pages')
+    const apiRoute = '/api/index'
+
+    const filePath = getSourceFileForPage(apiRoute, [pagesDir])
+
+    expect(filePath.replace(TEST_DIR, '')).toBe('/fixtures/page-extensions/default/pages/api/index/index.api.js')
```

Comment:
```
● getSourceFileForPage › handles getting file when index of folder

    expect(received).toBe(expected) // Object.is equality

    Expected: "/fixtures/page-extensions/default/pages/api/index/index.api.js"
    Received: "/fixtures/page-extensions/index/pages/api/index/index.js"

      245 |     const filePath = getSourceFileForPage(apiRoute, [pagesDir])
      246 |
    > 247 |     expect(filePath.replace(TEST_DIR, '')).toBe('/fixtures/page-extensions/default/pages/api/index/index.api.js')
          |                                            ^
      248 |   })
      249 | })
      250 |

      at Object.<anonymous> (test/helpers/files.spec.ts:247:44)
```
It seems like added test is not exactly right?

```suggestion
    expect(filePath.replace(TEST_DIR, '')).toBe('/fixtures/page-extensions/index/pages/api/index/index.js')
```